Browse location by repository facet

ives1227 opened the pull request:
## Description
On the Browse Locations results page, this creates a facet by the Location record Repository field to make it easier for a staff user to limit their view to desired records.  Also adds a 'None' option for the Repository.  

This allows a broader faceting option for the Locations.

## Related JIRA Ticket or GitHub Issue

## How Has This Been Tested?
This has been manually tested.  

## Screenshots (if appropriate):
<img width="433" alt="Screen Shot 2020-05-08 at 3 24 50 PM" src="https://user-images.githubusercontent.com/6585049/81441699-3d43a800-9140-11ea-9833-fc52027e1c08.png">

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ x] I have read the **CONTRIBUTING** document.
- [ x] I have authority to submit this code.
- [x ] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

mark-cooper commented:
Hi @ives1227 thanks for the pull request!

Unfortunately we're unable to review this one as it stands because of a merge conflict. Looking at the code at least some re-organization is needed around the search parts because the `SearchHelper` functionality was refactored (and `ExtraColumn` removed) in v2.8.0 as part of the user browse and search preferences work. Apologies for the inconvenience but if you're able to update the pull request we can aim to get it into the next release.

ives1227 commented:
I did a rebase and cleanup so this should be ready for the next release.  Thanks!

mark-cooper commented:
Hi @ives1227 thx for updating things. Unfortunately it looks like we've ended up with waaay too much 😄
Could you extract just the "Browse location by repository facet" feature into a new branch (or update this one)? When you've done that we can assign it for review!

ives1227 commented:
@mark-cooper - Yikes! You mean you don't want to review 900+ files? :)  I fixed that and it is down to one commit and 9 files.

mark-cooper commented:
@ives1227 that looks great, thx, much appreciated! Couple of things:

1. You mentioned having a selenium test to cover the new functionality but I don't see that included in the PR. Is that something you can add?

2. We will need a migration to reindex location records (for the `owner_repo`), but that's something we may want to defer for now. @lorawoodford you have an opinion on including a migration in this PR vs. us handling it later (anticipating other indexer related updates)?

lorawoodford commented:
Heya @ives1227 - Regarding @mark-cooper 's second point above, I'll go ahead and handle adding a narrowly scoped migration in the future/after this branch is merged.

So all that remains at this point is including any selenium tests (if you have any).

Best, 

Lora

ives1227 commented:
@mark-cooper We had a Selenium test written by another developer at one point but it got lost/broken in the different merges.  I have updated the PR and removed the bit about any Selenium tests.  It has been tested manually by end users.  

Also, just to clarify, @lorawoodford will take care of the migrations post merge.  Please let me know if this has changed.

mark-cooper commented:
@ives1227 thx for the update. That's correct on the migration -- one of us can handle it.

lorawoodford commented:
Thanks for your patiences on this, @ives1227 .  I'm going to go ahead and merge now, then I'll follow-up with a separate PR adding the migration and some tests.
